### DISCUSSION TITLE
Do we need to serve Octane (FrankenPHP) via Nginx?

### DISCUSSION THREAD

**Discussion by gazzoy** [laravel/framework#49495] [Q&A] [answered]
Hi artisans, I have a question about serving Octane (especially w/ FrankenPHP) via Nginx.

As documented, I believe it's recommended to run Octane via Nginx.
https://laravel.com/docs/10.x/octane#serving-your-application-via-nginx

However, I saw some articles say FrankenPHP doesnt require nginx anymore. So, my question is, if we run Octane w/ FrankenPHP, are we supposed to remove nginx or its still best practice to serve Octane via nginx even we use FrankenPHP?

Thanks for your help in advance.

**Comment by michael-rubel** [comment, marked as answer]
Absolutely not.

You can run the Octane application without Nginx. FrankenPHP is not required either. You can do that with Roadrunner or Swoole. Just bind the process on port 80 and (for example) configure the Cloudflare proxy to handle the frontend.

```php
php artisan octane:start --host=0.0.0.0 --port=80
```

Note: you'll probably need to send headers manually from your application.
For that purpose, you can use Laravel's built-in middleware. For example:

```php
Route::get('/', fn () => view('welcome'))
    ->name('home')
    ->middleware('cache.headers:public;max_age=31536000');
```

**Reply by gazzoy** [reply]
@michael-rubel Thanks really appericiate it. Will try that 👍
